scm/pipeline.go

```diff
@@ -0,0 +1,17 @@
+package scm
+
+import "time"
+
+type (
+	Pipeline struct {
+		ID          string     `json:"id"`
+		Status      string     `json:"status"`
+		Branch      string     `json:"branch"`
+		CommitSHA   string     `json:"commit_sha"`
+		Author      string     `json:"author"`
+		CreatedAt   time.Time  `json:"created_at"`
+		UpdatedAt   *time.Time `json:"updated_at,omitempty"`
```

Comment:
here are some fields from `PullRequest`
```
		Sha     string
		Ref     string
		Source  string
		Target  string
		Fork    string
		Link    string
		Diff    string
		Draft   bool
		Closed  bool
		Merged  bool
		Merge   string
		Base    Reference
		Head    Reference
		Author  User
		Created time.Time
		Updated time.Time
```

we should ensure `Pipeline` follows similar conventions.  For exmaple, `Sha` instead of `CommitSHA`, `Ref` instead of Branch (given pipelines can run for tags), `Created` instead of `CreatedAt`

Reply:
I also wonder if we should include the commit details / author / etc in the pipeline, since this is not really pipeline data it is execution data.  Also the commit details are already included in the hook

Reply:
Update the pipeline struct